snapcraft/internal/yaml.py

```diff
@@ -186,13 +190,16 @@ def _compute_part_dependencies(self):
                         found = True
                         break
                 if not found:
-                    wiki_part = self._wiki.get_part(dep)
-                    found = True if wiki_part else False
+                    remote_part = self._remote_parts.get_part(dep)
+                    found = True if remote_part else False
                     if found:
-                        plugin_name = wiki_part.pop('plugin')
+                        plugin_name = remote_part.pop('plugin')
                         part.deps.append(self.load_plugin(
-                            dep, plugin_name, wiki_part))
+                            dep, plugin_name, remote_part))
                         self._part_names.append(dep)
+                    else:
+                        logger.info('Maybe {!r} is defined as a remote part, '
+                                    'run `snapcraft update` to refresh')
```

Comment:
El 24/06/16 a las 03:04, Leo Arias escribió:
> In snapcraft/internal/yaml.py
> <https://github.com/ubuntu-core/snapcraft/pull/590#discussion_r68357000>:
>
> >                          self._part_names.append(dep)
> > +                    else:
> > +                        logger.info('Maybe {!r} is defined as a remote part, '
> > +                                    'run `snapcraft update` to refresh')
>
> also missing a test. And should this be an error, shouldn't it?
> This logic is so weird. I think this else should raise the exception.
> And I don't have it clear if in that case, the following statement is
> needed.
> Splitting smaller functions with nice names out of these loops would be
> nice, but again, maybe for a future refactor.

I couldn't agree more, tis is all horrible legacy code that needs some love (all of snapcraft.internal.yaml)

>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ubuntu-core/snapcraft/pull/590/files/a60b0459d698406fc3c67017a4c1009550f38566#r68357000>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABF_5ZWIuEVRP5YigYeqE-UeqU8CXCZVks5qO3N7gaJpZM4I7W40>.
>